presto-native-execution/presto_cpp/main/types/tests/FunctionMetadataTest.cpp

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <gtest/gtest.h>
+
+#include "velox/functions/prestosql/aggregates/RegisterAggregateFunctions.h"
+#include "velox/functions/prestosql/registration/RegistrationFunctions.h"
+#include "velox/functions/prestosql/window/WindowFunctionsRegistration.h"
+
+#include "presto_cpp/main/common/tests/test_json.h"
+#include "presto_cpp/main/types/FunctionMetadata.h"
+#include "presto_cpp/main/types/tests/TestUtils.h"
+
+using namespace facebook::velox;
+using namespace facebook::presto;
+
+using json = nlohmann::json;
+
+class FunctionMetadataTest : public ::testing::Test {
+ protected:
+  static void SetUpTestSuite() {
+    aggregate::prestosql::registerAllAggregateFunctions();
+    window::prestosql::registerAllWindowFunctions();
+    functions::prestosql::registerAllScalarFunctions();
+  }
+};
+
+TEST_F(FunctionMetadataTest, approxMostFrequent) {
+  std::string function = "approx_most_frequent";
```

Comment:
Write a test function in FunctionMetadataTest that takes as parameters the function name,  number of metadata and json file name and does the rest of this test body. The same logic is repeated in each test function.